src/main/java/com/conveyal/gtfs/loader/Table.java

```diff
@@ -226,20 +241,49 @@ public Table (String name, Class<? extends Entity> entityClass, Requirement requ
     ).addPrimaryKey();
 
     public static final Table STOPS = new Table("stops", Stop.class, REQUIRED,
-        new StringField("stop_id",  REQUIRED),
-        new StringField("stop_code",  OPTIONAL),
-        new StringField("stop_name",  REQUIRED),
-        new StringField("stop_desc",  OPTIONAL),
-        new DoubleField("stop_lat", REQUIRED, -80, 80, 6),
-        new DoubleField("stop_lon", REQUIRED, -180, 180, 6),
-        new StringField("zone_id",  OPTIONAL),
-        new URLField("stop_url",  OPTIONAL),
-        new ShortField("location_type", OPTIONAL, 2),
-        // FIXME: Need self-reference check during referential integrity check
-        new StringField("parent_station",  OPTIONAL), //.isReferenceToSelf()
-        new StringField("stop_timezone",  OPTIONAL),
-        new ShortField("wheelchair_boarding", OPTIONAL, 2)
-    ).restrictDelete().addPrimaryKey();
+        new StringField("stop_id", REQUIRED),
+        new StringField("stop_code", OPTIONAL),
+        new StringField("stop_name", OPTIONAL).requireConditions(),
```

Comment:
Add a TODO for all these empty requireConditions or implement logic.

Reply:
An optional list of conditions can be provided if needed. For example, with the stops table the stop_name, stop_lat, stop_lon and parent_station values are required for checking the conditions defined on the location_type field, but have no conditions on themselves.

Reply:
Ok, I see how this works. At this point though, there isn't any action that occurs within the `ReferenceTracker` class since an empty array of conditions will be analyzed. However, I suppose it might be useful if something at some point in the future checks if a specific field is conditionally required. I'd still add a comment though saying that the actual conditions that will be acted upon are within the `location_type` field.